Makefile

```diff
@@ -122,6 +120,8 @@ MMCTL_REL_TO_DOWNLOAD = $(shell scripts/get_latest_release.sh 'mattermost/mmctl'
 
 all: run ## Alias for 'run'.
 
+-include config.mk
+include default-config.mk
```

Comment:
It's not obvious to me why only four environment variables end up in `default-config.mk`, but not `BUILD_ENTERPRISE` or `GO` or `MMCTL_REL_TO_DOWNLOAD`.

Instead of a `default-config.mk`, could we just keep these definitions in this file (and above where they were originally located), and simply allow `config.mk` to override them? That gives `config.mk` the added advantage of being able to refer to the default values to make some decision.

Reply:
I would limit this to the settings that the user really wants to configure. For example, `BUILD_ENTERPRISE`, by default is inferred from the existence of the `BUILD_ENTERPRISE_DIR`. Is it a common use case to modify that variables? I can move almost all the variables there, but I don't want to make it a place where is hard to understand what to touch and what not.